src/ChluIPFS.js

```diff
@@ -121,6 +112,35 @@ class ChluIPFS {
         this.ipfs = undefined;
     }
 
+    async switchType(newType) {
+        if (this.type !== newType) {
+            await this.persistData();
+            if (this.type === constants.types.customer) {
+                if (this.db) await this.db.close();
+                this.db = undefined;
+            }
+            if (this.type === constants.types.service) {
+                if (this.room) {
+                    this.room.removeListener('message', this.serviceNodeRoomMessageListener);
+                    this.serviceNodeRoomMessageListener = undefined;
+                }
+                if (this.dbs) {
+                    await Promise.all(Object.values(this.dbs).map(db => db.close()));
+                }
+                this.dbs = {};
+            }
+            this.type = newType;
+            await this.loadPersistedData();
```

Comment:
At this point it just switched from the old type to the new type. However when it was created or switched to the old type, it loaded the data specific to that type. Now that it changed type it should load again because each type loads different data, that's also why it saves before switching. This is necessary to make sure nothing is lost

At line 116 there is a check that makes switching to the same type that the instance already is a noop